app/connectors/AddressLookupConnector.scala

```diff
@@ -16,121 +16,52 @@
 
 package connectors
 
-import javax.inject.Inject
 import com.codahale.metrics.{Counter, Timer}
 import common.Logging
 import config.WSHttp
+import javax.inject.{Inject, Singleton}
 import models.Address
 import models.external._
-import play.api.{Configuration, Environment}
+import play.api.Mode.Mode
 import play.api.i18n.MessagesApi
-import play.api.libs.json.{JsObject, Json}
-import play.api.mvc.Call
+import play.api.libs.json.Reads
+import play.api.{Configuration, Environment}
 import services.MetricsService
-import uk.gov.hmrc.http.{CoreGet, CorePost, HeaderCarrier, HttpResponse}
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
 import uk.gov.hmrc.play.config.ServicesConfig
 import uk.gov.hmrc.play.http.logging.MdcLoggingExecutionContext._
 
 import scala.concurrent.Future
 import scala.util.control.NoStackTrace
 
-class AddressLookupConnectorImpl @Inject()(val metricsService: MetricsService,
-                                           val messagesApi: MessagesApi,
-                                           val http: WSHttp,
-                                           override val runModeConfiguration: Configuration,
-                                           environment: Environment)extends AddressLookupConnector with ServicesConfig{
-  val addressLookupFrontendUrl     = baseUrl("address-lookup-frontend")
-  lazy val payeRegistrationUrl     = getConfString("paye-registration-frontend.www.url","")
-  val successCounter               = metricsService.addressLookupSuccessResponseCounter
-  val failedCounter                = metricsService.addressLookupFailedResponseCounter
-  def timer            = metricsService.addressLookupResponseTimer.time()
-  lazy val timeoutAmount: Int      = runModeConfiguration.underlying.getInt("timeoutInSeconds")
-  override protected def mode = environment.mode
-}
-
-class ALFLocationHeaderNotSetException extends NoStackTrace
-
-trait AddressLookupConnector extends Logging {
+@Singleton
+class AddressLookupConnector @Inject()(metricsService: MetricsService,
+                                       http: WSHttp,
+                                       override val runModeConfiguration: Configuration,
+                                       environment: Environment
+                                      )(implicit messagesApi: MessagesApi)
+  extends ServicesConfig with Logging {
 
-  val addressLookupFrontendUrl: String
-  val payeRegistrationUrl: String
-  val http: CoreGet with CorePost
-  val metricsService: MetricsService
-  val messagesApi: MessagesApi
+  override protected def mode: Mode = environment.mode
 
-  val successCounter: Counter
-  val failedCounter: Counter
-  def timer: Timer.Context
+  lazy val addressLookupFrontendUrl: String = baseUrl("address-lookup-frontend")
```

Comment:
I would put this in appConfig and inject it rather than extend ServicesConfig and overriding the run mode and environment